src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -403,6 +403,17 @@ export default function TransactionFeedV2() {
     [isFetching, data?.transactions, originalArgs?.endCursor, standByTransactions.confirmed]
   )
 
+  useEffect(
+    function updatePersistedFeedFirstPage() {
+      const isFirstPage = originalArgs?.endCursor === FIRST_PAGE_TIMESTAMP
+
+      if (isFirstPage && data?.transactions) {
+        dispatch(updateFeedFirstPage(data.transactions))
+      }
+    },
+    [data?.transactions, originalArgs?.endCursor]
+  )
```

Comment:
It seems we could skip this effect by reacting to the underlying RTK-Query API fetched action, within the transactions reducer, right?

Reply:
@jeanregisser I'll investigate this! RTK-Query dispatches `transactionFeedV2Api/executeQuery/fulfilled` action whenever the data fetch promise is successfully fulfilled but I'll need to look if there's a type-safe way of catching this action rather than:
```typescript
const TYPESAFE_FEED_DATA_RECEIVED_ACTION = "transactionFeedV2Api/executeQuery/fulfilled"

case TYPESAFE_FEED_DATA_RECEIVED_ACTION:
  return { ...state, ... }
```

But if there's no type-safe way - do you think this ^ approach would be sufficient enought?

Reply:
It seems like there is a type-safe way but it requires to use `createSlice`. Then we can use it as the following:
```typescript
 extraReducers: (builder) => {
    builder.addMatcher(
      transactionFeedV2Api.endpoints.login.matchFulfilled,
      (state, { payload }) => {
         // modify the state
      },
    )
  },
```

But for this we'll need to re-write the existing `transactions/reducer.ts` to slice.

Reply:
Thanks, moving it to to a slice would be good then.

Reply:
@jeanregisser addressed in [this PR](https://github.com/valora-inc/wallet/pull/6173/files)